### PR TITLE
ENH Versioned badge to elements

### PR DESCRIPTION
**Pull request by sabina-talipova**
## Description

New badge element was added in the title to show current state of changes of Element block. There are three different bages: 
- `Draft` for unpublished changes in new blocks;
- `Modified` for modified published blocks;

## Manual testing steps

- I create new Content Block in Elemental Area.
- I should see `Draft` badge.
- I populate Title field with "My title".
- I should see `Draft` badge.
- I save block.
- I should see `Draft` badge.
- I publish block.
- I should not see a badge.
- I update Title field.
- I should see `Modified` badge.
- I save block.
- I should see `Modified` badge.

## Issues

- https://github.com/silverstripe/silverstripe-elemental/issues/1124

## Pull request checklist

- [x] The target branch is correct
    - See [picking the right version](https://docs.silverstripe.org/en/contributing/code/#picking-the-right-version)
- [x] All commits are relevant to the purpose of the PR (e.g. no debug statements, unrelated refactoring, or arbitrary linting)
    - Small amounts of additional linting are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
- [x] The commit messages follow our [commit message guidelines](https://docs.silverstripe.org/en/contributing/code/#commit-messages)
- [x] The PR follows our [contribution guidelines](https://docs.silverstripe.org/en/contributing/code/)
- [x] Code changes follow our [coding conventions](https://docs.silverstripe.org/en/contributing/coding_conventions/)
- [x] This change is covered with tests (or tests aren't necessary for this change)
- [x] Any relevant User Help/Developer documentation is updated; for impactful changes, information is added to the changelog for the intended release
- [x] CI is green

### CODE REVIEW

File: tests/Behat/features/multiple-wysiwyg-configs.feature [hunk 1]
```diff
@@ -18,23 +18,23 @@ Feature: multiple elemental blocks with different HTMLEditorConfig instances
...
-    And I should see "Link to a file" in the ".mce-menu.mce-in" element
-    And I should see "Link to external URL" in the ".mce-menu.mce-in" element
-    And I should see "Anchor on a page" in the ".mce-menu.mce-in" element
-    And I should see "Link to email address" in the ".mce-menu.mce-in" element
+    And I click on the "#Form_ElementForm_2 button[aria-label^='Insert link']" element
+    Then I should see "Page on this site" in the ".tox-menu" element
+    And I should see "Link to a file" in the ".tox-menu" element
+    And I should see "Link to external URL" in the ".tox-menu" element
+    And I should see "Anchor on a page" in the ".tox-menu" element
+    And I should see "Link to email address" in the ".tox-menu" element
```
Comment (GuySartorelli) [inline tests/Behat/features/multiple-wysiwyg-configs.feature]: These changes probably need to be on 5.1 - I suspect that branch's CI will be failing without it.

Comment (sabina-talipova) [reply tests/Behat/features/multiple-wysiwyg-configs.feature]: Without these changes I get ` Element #Form_ElementForm_1 [aria-label^='Insert link'] button not found` in local env when I run tests even for branch `5`. 
But there is another issue with limited list of available link types in WIZIWIG.
multiple-wysiwyg-configs feature_35

Comment (GuySartorelli) [reply tests/Behat/features/multiple-wysiwyg-configs.feature]: > Without these changes I get  Element #Form_ElementForm_1 [aria-label^='Insert link'] button not found in local env when I run tests even for branch 5.

That would be resolved after merging these changes first into `5.1` (to resolve the failure there) and then merging _up_ to `5`. It's not actually related to the versioned badges work at all.

> But there is another issue with limited list of available link types in WIZIWIG.

That's also separate to this issue and should be raised and resolved separately (but is probably just that asset-admin and cms haven't merged up yet, if I had to guess)

Comment (sabina-talipova) [reply tests/Behat/features/multiple-wysiwyg-configs.feature]: Updated

**Comment by sabina-talipova** [comment]
> That would be resolved after merging these changes first into 5.1 (to resolve the failure there) and then merging up to 5. It's not actually related to the versioned badges work at all.

For failed behat tests I pushed [another PR](https://github.com/silverstripe/silverstripe-elemental/pull/1139)

> That's also separate to this issue and should be raised and resolved separately (but is probably just that asset-admin and cms haven't merged up yet, if I had to guess)

Yes, you're right this happens since new changes for elemental were already merged up, but not for `admin`, `asset-admin` and `cms`. So it should be fixed automatically.

**Comment by sabina-talipova** [comment]
Failed tests require [another PR](https://github.com/silverstripe/silverstripe-elemental/pull/1139)

**Comment by emteknetnz** [comment]
@sabina-talipova I've merged up the other PR from 5.1 to 5, though you may need to rebase this PR on 5 in order for behat to pass

**Review by emteknetnz — APPROVED** [review]
Behat failure is unrelated

**Pull request merged**